frontend_tests/zjsunit/namespace.js

```diff
@@ -19,28 +21,57 @@ exports.patch_builtin = function (name, val) {
     return val;
 };
 
-exports.zrequire = function (name, fn) {
+function resolve_name(name, fn) {
     if (fn === undefined) {
         fn = '../../static/js/' + name;
     } else if (/generated\/|js\/|third\//.test(fn)) {
         // FIXME: Stealing part of the NPM namespace is confusing.
         fn = '../../static/' + fn;
     }
-    delete require.cache[require.resolve(fn)];
-    var obj = require(fn);
-    requires.push(fn);
+    return fn;
+}
+
+// Require a module and get a handle to it for monkey patching.
+// Can not patch if module.exports is not an object (only objects are passed by
+// reference) or the module is imported with ES6's import statement (it has a
+// separate cache).
+// Put this before other zrequire-s if you run into caching issues.
+exports.zrequire_pure = function (name, fn) {
+    fn = resolve_name(name, fn);
+    const path = require.resolve(fn);
+    require(fn);
+    return require.cache[path].exports;
```

Comment:
Is that different from `return require(fn)`?

Reply:
Huh I did not expect that to work for stubbing!